lib/srv/sess.go

```diff
@@ -995,16 +1011,12 @@ func (s *session) startInteractive(ch ssh.Channel, ctx *ServerContext) error {
 		return trace.ConvertSystemError(err)
 	}
 
-	if err := s.addParty(p, types.SessionPeerMode); err != nil {
-		return trace.Wrap(err)
-	}
-
 	// Open a BPF recording session. If BPF was not configured, not available,
 	// or running in a recording proxy, OpenSession is a NOP.
 	s.bpfContext = &bpf.SessionContext{
 		Context:   ctx.srv.Context(),
 		PID:       s.term.PID(),
-		Emitter:   s.recorder,
+		Emitter:   s.Recorder(),
```

Comment:
How will enhanced session recording respond to the recorder being closed? Should it be included in the `best-effort` approach?

Reply:
For best-effort scenarios, it is not affected. The only thing is that it will still try to emit events even after the session is entered in "no recording mode", producing warning messages during the entire session.

The session will still be closed in strict mode when emitting data events (which still happen even when the session has enhanced recording).